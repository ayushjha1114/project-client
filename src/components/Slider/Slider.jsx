import React from 'react';
import PropTypes from 'prop-types';
import { DEFAULT_BANNER_IMAGE } from '../../configs/constants';
import { getRandomNumber, getNextRoundRobin } from '../../lib/utils';

const propTypes = {
  altText: PropTypes.string,
  banners: PropTypes.arr,
  defaultBanner: PropTypes.string,
  duration: PropTypes.number,
  height: PropTypes.number,
  random: PropTypes.bool,
};

// default values for props:
const defaultProps = {
  altText: 'Default Banner',
  banners: '',
  defaultBanner: DEFAULT_BANNER_IMAGE,
  duration: 2000,
  height: 300,
  random: false,
};

class Slider extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      index: 0,
    };
  }

  componentDidMount() {
    const { random, duration } = this.props;
    this.interval = setInterval(() => {
      const { index } = this.state;
      if (random) {
        this.setState({
          index: getRandomNumber(5),
        });
        return;
      }
      const val = getNextRoundRobin(4, index);
      this.setState({
        index: val,
      });
    }, duration);
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  render() {
    const {
      altText,
      banners,
      random,
      defaultBanner,
      height,
      ...rest
    } = this.props;
    const { index } = this.state;
    const source = (banners) ? banners[index] : defaultBanner;
    return (
      <>
        <div style={{ textAlign: 'center' }}>
          <img src={source} {...rest} alt={altText} height={height} />
        </div>
      </>
    );
  }
}

Slider.propTypes = propTypes;
Slider.defaultProps = defaultProps;

export default Slider;
