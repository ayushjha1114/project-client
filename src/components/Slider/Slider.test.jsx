import React from 'react';
import { shallow, mount } from 'enzyme';
import Slider from './Slider';

describe('Slider', () => {
  test('should render correctly', () => {
    const component = shallow(<Slider />);
    expect(component).toMatchSnapshot();
  });
  it('check options props', () => {
    const props = {
      options: ['asasa'],
    };

    const component = mount(<Slider {...props} />);
    expect(component.prop('options')).toEqual(['asasa']);
  });
});
