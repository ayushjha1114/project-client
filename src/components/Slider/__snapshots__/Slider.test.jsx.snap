// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Slider
    altText="Default Banner"
    banners=""
    defaultBanner="images/default.png"
    duration={2000}
    height={200}
    random={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <img
          alt="Default Banner"
          duration={2000}
          height={200}
          src="images/default.png"
        />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="Default Banner"
          duration={2000}
          height={200}
          src="images/default.png"
        />,
        "style": Object {
          "textAlign": "center",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Default Banner",
          "duration": 2000,
          "height": 200,
          "src": "images/default.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <img
            alt="Default Banner"
            duration={2000}
            height={200}
            src="images/default.png"
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="Default Banner"
            duration={2000}
            height={200}
            src="images/default.png"
          />,
          "style": Object {
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Default Banner",
            "duration": 2000,
            "height": 200,
            "src": "images/default.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
